### ISSUE TITLE
[feature]: fast `make unit` on Apple M1

### ISSUE THREAD

**Issue by joostjager** [lightningnetwork/lnd#7084]
It seems that `make unit` is unworkably slow on Apple M1 machines. Potentially because of https://github.com/etcd-io/bbolt/issues/149.

What fixes it for me is to run `make unit` inside a simple docker container.

This issue is about adding a new make target for `unit-in-container` to make fast unit tests easy.

**Comment by Roasbeef** [comment]
How long does it take for you? 

As is, afaict, it actually runs all the tests serially, to ensure that it's testing all the internal packages and migrations, etc. 

If you do a normal `go test -v ./...` it'll start to run them in parallel, but it'll end up missing some modules.

**Comment by joostjager** [comment]
Hmm, that command is similar for me, 10:47.56 total

What I had been running and didn't work for me was `make unit pkg=htlcswitch`. It just times out, and it does work inside docker.

**Comment by positiveblue** [comment]
In my laptop (linux 5.15, 8 core AMD 5850U , 32Gb RAM), running ` go clean -testcache ./...` between steps so go does not skip cached tests:

```
make unit  622.63s user 118.20s system 106% cpu 11:37.09 total
go test -v ./...  420.96s user 37.29s system 404% cpu 1:53.41 total # got failures in htlcswitch & htlcswitch/hodl
make unit pkg=htlcswitch  198.12s user 7.37s system 183% cpu 1:51.76 total
```

**Comment by joostjager** [comment]
Yes, that is roughly what I am seeing on an AMD ryzen machine too. On M1, it times out though. Unless I run in docker.

**Comment by joostjager** [comment]
This is a relevant thread too: https://twitter.com/marcan42/status/1494213862970707969?lang=en

It almost looks like running code in a docker on macos bypasses a proper fsync. And this makes unit tests fast :(

**Comment by saubyk** [comment]
Closing as not a dev priority for the team at the moment. Not seen any prs on this as well.
